src/mutual_information_sensitivity.jl

```diff
@@ -0,0 +1,272 @@
+
+### First Order Sensitivity Indices
+The first order sensitivity indices are calculated as the mutual information between the input ``X_i`` and the output ``Y`` divided by the entropy of the output ``Y``:
+
+```math
+S_{1,i} = \frac{I(X_i;Y)}{H(Y)}
+```
+
+This measure is introduced in Lüdtke et al. (2007)[^1] and also present in Datseris & Parlitz (2022)[^2] in an unnormalized form.
+
+### Second Order Sensitivity Indices
+To account for the interaction between input parameters, and their effect on the output, the second order sensitivity indices can be calculated using the 
+conditional mutual information between two input parameters ``X_i`` and ``X_j`` given the output ``Y``. They have to be corrected for the correlation between the input parameters. The
+second order sensitivity indices according to Lüdtke et al. (2007)[^1] are given by:
+
+```math
+S_{2,i} = \frac{I(X_i;X_j|Y) - I(X_i;X_j)}{H(Y)}
+```
+
+### Total Order Sensitivity Indices
+From Lüdtke et al. (2007)[^1], the total order sensitivity indices can be calculated as:
+
+```math
+S_{\text{total},i} = \frac{H(Y) - H(Y|\{X_1,...,X_n\} \\ X_i)}{H(Y) - H_{\Delta}}}
+```
+
+Where ``H_{\Delta}`` is the discretization entropy of the output, which is introduced to account for the discretization of the input space. 
+
+## API
+
+    gsa(f, method::MutualInformation, p_range; samples, batch = false)
+
+Returns a `MutualInformationResult` object containing the resulting sensitivity indices for the parameters and the corresponding confidence intervals.
+The `MutualInformationResult` object contains the following fields:
+- `S1`: First order sensitivity indices.
+- `S1_Conf_Int`: Confidence intervals for the first order sensitivity indices.
+- `S2`: Second order sensitivity indices.
+- `S2_Conf_Int`: Confidence intervals for the second order sensitivity indices.
+- `ST`: Total order sensitivity indices.
+- `ST_Conf_Int`: Confidence intervals for the total order sensitivity indices.
+
+For fields that are not calculated, the corresponding field in the result will be an array of zeros.
+
+### Example
+
+```julia
+using GlobalSensitivity
+
+function ishi_batch(X)
+    A= 7
+    B= 0.1
+    @. sin(X[1,:]) + A*sin(X[2,:])^2+ B*X[3,:]^4 *sin(X[1,:])
+end
+function ishi(X)
+    A= 7
+    B= 0.1
+    sin(X[1]) + A*sin(X[2])^2+ B*X[3]^4 *sin(X[1])
+end
+
+lb = -ones(4)*π
+ub = ones(4)*π
+
+res1 = gsa(ishi,MutualInformation(),[[lb[i],ub[i]] for i in 1:4],samples=15000)
+res2 = gsa(ishi_batch,MutualInformation(),[[lb[i],ub[i]] for i in 1:4],samples=15000,batch=true)
+```
+
+### References
+[^1]: Lüdtke, N., Panzeri, S., Brown, M., Broomhead, D. S., Knowles, J., Montemurro, M. A., & Kell, D. B. (2007). Information-theoretic sensitivity analysis: a general method for credit assignment in complex networks. Journal of The Royal Society Interface, 5(19), 223–235.
+[^2]: Datseris, G., & Parlitz, U. (2022). Information Across Timeseries. 105-119.
```

Comment:
```suggestion
[^2]: Datseris, G., & Parlitz, U. (2022). Nonlinear Dynamics, Ch. 7, pg. 105-119.
```

I think it will be easier for people to find this if you cite the book name directly. Thank you for citing us!!!